app/src/sailfish-qml/qml/harbour-musikloud2.qml

```diff
@@ -14,7 +15,52 @@ ApplicationWindow
     AudioPlayer {
         id: player
 
-        onStatusChanged: if (status == AudioPlayer.Failed) infoBanner.showError(errorString);
+        property SoundCloudTrackModel searchResults
+
+        onStatusChanged: {
+            if (status == AudioPlayer.Failed) infoBanner.showError(errorString);
+        }
+
+        onCurrentIndexChanged: {
+            // Fetch more tracks in current tracks to play are all played
+            var model = player.getSearchResults();
+            if (model && (player.currentIndex + 1) >= model.rowCount()) {
+                console.log("Fetching more tracks")
+                model.fetchMore()
+            }
+        }
+
+        function setTracksToPlay(trackModel) {
```

Comment:
Okay then just loop in the _addFetchResults function instead of clearing the queue. It doesn't seem nice to clear the queue if we really just add stuff to it